schemas/expconf/v0/hyperparameters.json

```diff
@@ -18,6 +18,13 @@
         }
     },
     "additionalProperties": {
-        "$ref": "http://determined.ai/schemas/expconf/v0/hyperparameter.json"
+        "anyOf": [
+            {
+                "$ref": "http://determined.ai/schemas/expconf/v0/hyperparameter.json"
+            },
+            {
+                "$ref": "http://determined.ai/schemas/expconf/v0/hyperparameters.json"
```

Comment:
This will _almost_ never be valid, because it looks to me like each nesting would have to have a `global_batch_size` for this to be true.

I think the recursion needs to be added inside of `hyperparameter.json`

Fun fact: I originally planned on landing nested hp as part of expconf when I started.

Other fun fact, I just realized that I _forgot to fully remove nested hp from expconf_.  You can see that `hyperparameter.json` _already has a recursion_.

If you want to also allow lists in nested hp (which seems only logical to me), you can do that with the following additional diff to `hyperparameter.json`:

```diff
             {
                 "unionKey": "const:type=categorical",
                 "$ref": "http://determined.ai/schemas/expconf/v0/hyperparameter-categorical.json"
             },
+            {
+                "unionKey": "type:array",
+                "type": "array",
+                "items": {
+                    "$ref": "http://determined.ai/schemas/expconf/v0/hyperparameter.json"
+                }
+            },
             {
                 "unionKey": "always",
                 "type": "object",
                 "checks": {
                     "if a hyperparameter object's [\"type\"] is set, it must be one of \"int\", \"double\", \"log\", const\", or \"categorical\"": {
                         "properties": {
                             "type": false
                         }
                     }
                 },
                 "additionalProperties": {
                     "$ref": "http://determined.ai/schemas/expconf/v0/hyperparameter.json"
                 }
             },
             {
                 "unionKey": "never",
                 "not": {
-                    "type": "object"
+                    "type": [
+                        "object",
+                        "array"
+                    ]
                 }
             }
```

(that part I wrote a long time ago and did remember to remove)